torch/csrc/jit/tensorexpr/loopnest.h

```diff
@@ -233,10 +233,49 @@ class TORCH_API LoopNest {
   // the temporary buffer used in the computation.
   void computeAt(Stmt* s, For* at);
 
-  void rfactor(
-      const Expr* f,
-      const Var* reduction_var,
-      Block* insertion_point = nullptr /* optional */);
+  // Rfactor a reduction axis into a normal axis.
+  //
+  // Requirements:
+  //  * S is the reduction store
+  //  * S is the only statement in the innermost loop
+  //  * There is at least two reduction arguments in S
+  //  * OUTER_REDUCTION_FOR loop corresponds to the outermost reduction variable
+  //  used in the store and all other reduction variables are index variables of
+  //  children loops of OUTER_REDUCTION_FOR
+  //  * OUTER_REDUCTION_FOR is a perfect loop nest, i.e. it has only loops
+  //  corresponding to the other reduction variables and the store, nested into
+  //  each other
+  //
+  // What it does:
+  //   * Introduce a new buffer with an extra dimension of a size equal to the
+  //   span of the loop OUTER_REDUCTION_FOR (the new buffer is returned via
+  //   RFAC_BUF_PTR)
+  //   * Insert an initialization store for the new buffer in
+  //   OUTER_REDUCTION_FOR before its nested loop
```

Comment:
This is not in the example below, right?

Reply:
Oh, good catch, I think I made a mistake in the example